crates/solver/src/solver/single_order_solver.rs

```diff
@@ -336,113 +450,9 @@ enum SolveResult {
-                    SolveResult::Solved(settlement) => {
-                        let _ = tx.send(settlement);
-                    }
-                    SolveResult::RateLimited => {
-                        tracing::warn!(
-                            solver = %self.name(),
-                            "rate limited; backing off until next auction"
-                        );
-                        break;
-                    }
-                }
-            }
-            drop(tx);
-        };
-
-        let finalize = async {
-            let mut index = 0;
-            while let Some(intermediate) = rx.recv().await {
-                match self
-                    .finalize_settlement(intermediate, &external_prices, gas_price, index)
-                    .await
-                {
-                    Ok(Some(settlement)) => {
-                        settlements.push(settlement);
-                    }
-                    Ok(None) => (),
-                    Err(err) => {
-                        tracing::warn!(?err, "failed to finalize intermediate settlement");
-                    }
-                }
-                index += 1;
-            }
-        };
-
-        // Subtract a small amount of time to ensure that the driver doesn't reach the
-        // deadline first.
-        let timeout =
-            tokio::time::sleep_until(deadline.checked_sub(Duration::from_secs(1)).unwrap().into());
-
-        // open new scope for solve->finalize pipeline to make borrow checker happy
-        {
-            let solve = solve.fuse();
-            futures::pin_mut!(solve);
-            futures::pin_mut!(finalize);
-            futures::pin_mut!(timeout);
-            loop {
-                tokio::select! {
-                    // if `solve` stops early there are still settlements to be finalized
-                    _ = &mut solve => (),
-                    // all possible settlements have been finalized
-                    _ = &mut finalize => break,
-                    // we reached the timeout and should return the results ASAP
-                    _ = &mut timeout => break,
-                }
-            }
-        }
-
-        self.fills.collect_garbage();
-
-        // Shuffle so that in the case a buggy solver keeps returning some amount
-        // of invalid settlements first we have a chance to make progress.
-        settlements.shuffle(&mut rand::thread_rng());
-        // Keep at most this many settlements to not overwhelm the node with too many
-        // simulations.
-        settlements.truncate(self.max_settlements_per_solver);
-
-        merge::merge_settlements(
-            self.max_merged_settlements,
-            &external_prices,
-            &mut settlements,
-        );
-
-        Ok(settlements)
+    async fn solve(&self, auction: Auction) -> Result<Vec<Settlement>> {
+        let span = tracing::info_span!("", solver = self.name());
```

Comment:
ubernit: can we add a span name as well?
```suggestion
        let span = tracing::info_span!("solve", solver = self.name());
```

Reply:
Actually, come to think of it - why don't we create a `solve` span for every solver (including external solvers) that we call. That is, istead of opening a span for single order solvers, you can open one in:

https://github.com/cowprotocol/services/blob/8352664037bb98bce70fefb4d7ba555c979fa007/crates/solver/src/driver.rs#L181